File: go/6kyu/bitCounting.go
package main

import "fmt"

func main() {
	fmt.Println(CountBits(1234))
}

func CountBits(n uint) int {
	var i int = 0
	for n > 0 {
		i = i + int(n)%2
		n = n / 2
	}
	return i
}

File: javascript/6kyu/bitCounting.js
var countBits = function(n) {
  return +n.toString(2).split("").reduce((prev, cur) => +prev + +cur)
};

console.log(countBits(0)) // 0
console.log(countBits(4)) // 1
console.log(countBits(7)) // 3
console.log(countBits(9)) // 2
